### ISSUE TITLE
installation troubles for China based server

### ISSUE THREAD

**Issue by erjenkins29** [commercialhaskell/stack#4349]
Two issues: 

1. I was trying to upgrade my original stack v 1.5 using stack upgrade, and ran into an error which was resolved by stack 1.6.  So I tried uninstalling the original stack and just reinstalling the new version.  However, when I tried, this happened:

```
[jupyter@Server ~]$ wget -qO- https://get.haskellstack.org/ | sh
Detected Linux distribution: centos

[sudo] password for jupyter:
Loaded plugins: fastestmirror, refresh-packagekit, security
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirrors.cn99.com
 * epel: mirrors.huaweicloud.com
 * extras: mirrors.cn99.com
 * updates: ftp.sjtu.edu.cn
Package 4:perl-5.10.1-144.el6.x86_64 already installed and latest version
Package 1:make-3.81-23.el6.x86_64 already installed and latest version
Package automake-1.11.1-4.el6.noarch already installed and latest version
Package gcc-4.4.7-23.el6.x86_64 already installed and latest version
Package gmp-devel-4.3.1-13.el6.x86_64 already installed and latest version
Package libffi-3.0.5-3.2.el6.x86_64 already installed and latest version
Package zlib-1.2.3-29.el6.x86_64 already installed and latest version
Package xz-4.999.9-0.5.beta.20091007git.el6.x86_64 already installed and latest version
Package 2:tar-1.23-15.el6_8.x86_64 already installed and latest version
Package matching git-1.7.1-9.el6_9.x86_64 already installed. Checking for update.
Package gnupg2-2.0.14-9.el6_10.x86_64 already installed and latest version
Nothing to do

Using generic bindist...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
103   621    0   621    0     0    408      0 --:--:--  0:00:01 --:--:--  1556

```
This didn't work until about the 15th attempt.  The cause is almost certainly because I'm located in China, where downloading things like this is always a huge headache (connection to github, for example, can be spotty).  This was my experience with the Chinese mirrors... I resolved it by downloading manually, but would be nice if the curl/wget method worked faster.  That's my first issue.

I then ran into an issue because I was using centos 6, but I've made a separate issue for that here: #4348 

2. ANYONE living in china must add the following to their ~/.stack/config.yaml:

```
### ADD THIS IF YOU LIVE IN CHINA
setup-info: "http://mirrors.tuna.tsinghua.edu.cn/stackage/stack-setup.yaml"
urls:
  latest-snapshot: http://mirrors.tuna.tsinghua.edu.cn/stackage/snapshots.json
  lts-build-plans: http://mirrors.tuna.tsinghua.edu.cn/stackage/lts-haskell/
  nightly-build-plans: http://mirrors.tuna.tsinghua.edu.cn/stackage/stackage-nightly/
package-indices:
  - name: Tsinghua
    download-prefix: http://mirrors.tuna.tsinghua.edu.cn/hackage/package/
    http: http://mirrors.tuna.tsinghua.edu.cn/hackage/00-index.tar.gz

```
the source for that is: https://mirror.tuna.tsinghua.edu.cn/help/hackage/

There are ways to find this information out, but if you don't speak Chinese, it can be very difficult to find.  So this is just a PSA.  My assumption is that it is literally impossible to use stack from china otherwise, unless you equip your server with a vpn.

**Comment by mattaudesse** [comment]
This is really helpful @erjenkins29 - thanks.

Perhaps it'd be good to add this under a new section in the [docs](https://docs.haskellstack.org/en/stable/yaml_configuration/#non-project-specific-config). If you're up for submitting another PR that'd be much appreciated.

**Comment by mattaudesse** [comment]
Sounds good - thanks!

**Comment by snoyberg** [comment]
Close like this was addressed by #4376, closing
